### ISSUE TITLE
[BUG]Reprap Discount Full Graphics Smart Controller Display is Garbled

### ISSUE THREAD

**Issue by roland1970** [MarlinFirmware/Marlin#15391]
Hello

I have downloaded 2.0 bugfix ,my display is an Reprap Discount Full Graphics Smart Controller Display .
Problem with the display is that is garbled after flashing the firmware.
That problem did I also have with version 1.9 and that is solved by the following instruction.
```
U8GLIB_ST7920_128X64_RRD u8g(0); // Number of stripes can be adjusted in ultralcd_st7920_u8glib_rrd.h with PAGE_HEIGHT              

 U8GLIB_ST7920_128X64_1X u8g(23,17,16); // Number of stripes can be adjusted in ultralcd_st7920_u8glib_rrd.h with PAGE_HEIGHT
```

The codes located in ultralcd_impl_DOGM.h
The first line did I disabled en copy and adjusted in the second line .
So did I fix that the display works .
Did anyone a suggestion how to fix my problem in 2.0 ?

Printer is a Hypercube Evo with Arduino en Ramps board.

**Comment by ALEX-45** [comment]
Yes, there is a problem. But there is no file "ultralcd_impl_DOGM.h", and I don’t know how to fix it myself.

in 2.0

**Comment by ALEX-45** [comment]
recent bug, but a very important issue. I observe in 2.0 and in 1.1.x

**Comment by Coollision** [comment]
just spent some hours on it as wel, found a fix for marlin 2.0, in the file src/lcd/dogm/ultralcd_DOGM.h on line 54 (you'l see somewhere above that line a comment saying RepRap Discount Full Graphics Smart Controller) change the line from #define U8G_CLASS U8GLIB_ST7920_128X64_RRD
to #define U8G_CLASS U8GLIB_ST7920_128X64_1X
that worked for me

**Comment by ALEX-45** [comment]
I dealt with a bug adding the following lines in the configuration.h

//my add исправление дисплея https://3dtoday.ru/questions/artefakty-na-lcd-12864
#define ST7920_DELAY_1 DELAY_NS(0)
#define ST7920_DELAY_2 DELAY_NS(188) 
#define ST7920_DELAY_3 DELAY_NS(0)

It was like this:
![photo_2019-10-09_16-29-48](https://user-images.githubusercontent.com/26367411/67043610-22253280-f133-11e9-91da-c3a773c5bd86.jpg)

Fixed.

**Comment by Coollision** [comment]
Indeed forgot to mention that, did that as well.
Delay 1 and 2 I've set to 20 and 3 to 70

**Comment by boelle** [comment]
@Coollision maybe submit a PR so that everyone can benefit?

**Comment by InsanityAutomation** [comment]
@boelle that doesnt make sense to do. The defaults work for most. When you get something that it doesnt it needs adjusting.

**Comment by boelle** [comment]
just a small brainfart from my side

so this is an "edge" case? if so should we just close this one or is there a better action to take?

**Comment by ALEX-45** [comment]
in theory, you need to make a choice from macro definitions for different displays (if they are different) so that the firmware works out of the box. In the old versions everything worked and so, and in the new ones the timings were apparently cut. Thus, there is a bug, and at the moment it is fixed only with a crutch, since most ordinary users can only uncomment or comment on definitions and change numerical values in the configuration file.

**Comment by Coollision** [comment]
I think indeed the timing are there to adjust on a per user basis, but perhaps someone with more experience could look into my adjustment of the ultralcd_DOGM.h file?

**Comment by boelle** [comment]
@shitcreek time to have a quick look ?

**Comment by viebvo** [comment]
Just wanted to add that i had the same problem with current 2.0 and the fix from @Coollision worked for me.

**Comment by Coollision** [comment]
@viebvo only the fix of the src/lcd/dogm/ultralcd_DOGM.h or did you adjust the timings as well?

**Comment by viebvo** [comment]
@Coollision - just changed from "#define U8G_CLASS U8GLIB_ST7920_128X64_RRD" to "#define U8G_CLASS U8GLIB_ST7920_128X64_1X"
The file ultralcd_impl_DOGM.h is missing in the current 2.0 sourcetree.

**Comment by boelle** [comment]
@InsanityAutomation would it be fair to close this one as it could fall under the per user case?

**Comment by MechyJoel** [comment]
Roland1970 suggested fix worked for me as well. Had to search ultralcd in the marlin folder to find the appropriate .h file to edit.

**Comment by Whitke** [comment]
I know this is closed, but just wanted to thank the group for the info! @Coollision your formula worked to clean up my LCD screen. 
Thank you!

**Comment by zalexzperez** [comment]
> I dealt with a bug adding the following lines in the configuration.h
> 
> //my add исправление дисплея https://3dtoday.ru/questions/artefakty-na-lcd-12864
> #define ST7920_DELAY_1 DELAY_NS(0)
> #define ST7920_DELAY_2 DELAY_NS(188)
> #define ST7920_DELAY_3 DELAY_NS(0)
> 
> It was like this:
> ![photo_2019-10-09_16-29-48](https://user-images.githubusercontent.com/26367411/67043610-22253280-f133-11e9-91da-c3a773c5bd86.jpg)
> 
> Fixed.

Fixed my Ender3 on the latest bugfix 2.0.x nighlty compilation. Thank you!

**Comment by ALEX-45** [comment]
> Fixed my Ender3 on the latest bugfix 2.0.x nighlty compilation. Thank you!

not at all
